src/CommunityToolkit.Maui.Core/Interfaces/IDockLayout.shared.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace CommunityToolkit.Maui.Core.Interfaces;
+
+/// <summary>Docking position for a view in <see cref="IDockLayout"/>.</summary>
+public enum DockEnum
+{
+	/// <summary>Do not dock.</summary>
+	None = 0,
+	/// <summary>Dock at the top.</summary>
+	Top = 1,
+	/// <summary>Dock to the left.</summary>
+	Left = 2,
+	/// <summary>Dock to the right.</summary>
+	Right = 3,
+	/// <summary>Dock at the bottom.</summary>
+	Bottom = 4
+}
```

Comment:
Crazy thought - what if we make this enum a flag by adding the `[Flags]` attribute?

Then, for example, users could place a Dock on both the top + bottom:

var dockLayout = new DockLayout();
dockLayout.SetPosition(DockPosition.Top | DockPosition.Bottom);

I imagine most devs won't use it as a flag, but implementing the enum as a Flag now is easy whereas if we ever wanted to change it later would be a breaking change.


```suggestion
[Flags]
public enum DockEnum
{
	/// <summary>Do not dock.</summary>
	None = 0,
	/// <summary>Dock at the top.</summary>
	Top = 1,
	/// <summary>Dock to the left.</summary>
	Left = 2,
	/// <summary>Dock to the right.</summary>
	Right = 4,
	/// <summary>Dock at the bottom.</summary>
	Bottom = 8
}
```

Reply:
But how this works in the switch? I believe we will need to add a case for the `Top | Bottom` case, no?

Reply:
Good point! We'd convert the `switch` to a series of `if` statements:

```cs
if(position is DockPosition.Top)
{
  // ...
}
if(position is DockPosition.Bottom)
{
  // ...
}
if(position is DockPosition.Left)
{
  // ...
}
if(position is DockPosition.Right)
{
  // ...
}

Reply:
> Then, for example, users could place a Dock on both the top + bottom:

Not sure, I get your point. DockPosition (or DockEnum) refers to the position of the child view inside the DockLayout container. It's not `dockLayout.SetPosition(DockPosition.Top | DockPosition.Bottom);`, but `DockLayout.SetPosition(childView, DockPosition.Top | DockPosition.Bottom);`. How can you dock a view both top and bottom at the same time? What does that mean? Do you think in terms of 'anchoring' the view? If so, that would also drastically change the layout calculations.

Reply:
I agree with @profix898 . We can add additional values like TopLeft or BottomRight, but the flag is not correct, because it allows Top | Bottom. Anyway it won't be a breaking change, so we can enhance it by request.